src/DeviceInterfaceValues.tsx

```diff
@@ -386,48 +392,75 @@ const SendInterfaceDataModal = ({
           {selectedMapping && (
             <Form.Group as={Col} controlId="formValue">
               <Form.Label className="m-0">Please enter the value:</Form.Label>
-              <Form.Control
-                type={
-                  selectedMapping.type === 'integer' ||
-                  // TODO: Long integer should not be number, validate them with BigInt, and send them to Astarte as strings.
-                  selectedMapping.type === 'longinteger' ||
-                  selectedMapping.type === 'double'
-                    ? 'number'
-                    : 'text'
-                }
-                value={value}
-                onChange={handleValueChange}
-                isInvalid={!!errors.value}
-              />
+              {selectedMapping.type === 'boolean' ? (
+                <Form.Select
+                  value={value.toString()}
+                  onChange={handleBooleanValueChange}
```

Comment:
behind the sceene we're still handling input as text. We can still use `handleValueChange`